### DISCUSSION TITLE
endless integrations with all python framworks

### DISCUSSION THREAD

**Discussion by ArtemIsmagilov** [strawberry-graphql/strawberry#3471] [Q&A] [answered]
Explain to me why developers make integrations with other frameworks? for example, graphene integrated with sqlalchemy, django, flask, and so on. However, sqlalchemy does not integrate with anyone. after all, a good library is one that does not create unnecessary dependencies but simply fits perfectly with any technology stack and does not require integrations.
Is it really impossible to make a good single integration for any frameworks. Example sqlalchemy.

**Comment by ArtemIsmagilov** [comment]
I can assume that integrations simplify the development cycle https://www.reddit.com/r/graphql/comments/mzrfod/strawberry_vs_graphene/

**Comment by patrick91** [comment, marked as answer]
@ArtemIsmagilov I converted this to a discussion, hope that's ok 😊

I think different people might have different opinions on this, so here's my view on it.

First of all, there's different kinds of integrations, at least for Strawberry. We have integrations on the HTTP layer (Django, Flask, FastAPI and so on) and integrations on the ORM/Model layer (Django, Pydantic, SQLAlchemy)

I think the ones for the HTTP layer should pretty much be provided by us, we want a great developer experience, and providing the HTTP layer is one of the main things we should do. Imagine having to implement all the code for dealing with the GraphQL request/response cycle, also because it is code that is needed in 99% of the projects 😊

For the model layer, I see it as a convenience for some use cases, being able to quickly create a GraphQL API on top of existing code ([Strawberry Django](https://github.com/strawberry-graphql/strawberry-django) has a lot of niceties, for example other than converting Django Models to GraphQL types, you also get automatic SQL optimisation, which is neat 😊

I don't understand this point though:

> a good library is one that does not create unnecessary dependencies but simply fits perfectly with any technology stack and does not require integrations.

Strawberry doesn't have unnecessary dependencies, right?

**Reply by ArtemIsmagilov** [reply]
@patrick91, I thank you for your very detailed and extensive answer. Now I have no questions about this. Developers really need to manage complexity and large code. Thanks again for answering. It seems that now I will be asking questions not to ChatGPT, but to the current developers here. Well, on a slightly different topic - it wouldn’t be bad if they added SCHEMA-FIRST, this would allow you to initialize both SCHEMA-FIRST and CODE-FIRST schemes. Why not?)

**Reply by ArtemIsmagilov** [reply]
Although my proposal for the scheme already contradicts the Zen of Python
